var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -297,6 +305,11 @@ def enable_or_disable(variant, keyword='USE', var=None, newer=False):
         enable_or_disable('zstd', newer=True)
         enable_or_disable('tbb', newer=True)
 
+    @run_before('install')
+    def build_amd(self):
+        if '+rocm' in self.spec:
+            python(os.path.join('tools', 'amd_build', 'build_amd.py'))
```

Comment:
This should probably be part of the build process not a pre-install step

Reply:
PyTorch is a bit of an odd case here. Normally, `PythonPackage` builds have 2 stages: build and install. For PyTorch (and many other Python libraries), when you run `python setup.py build` and `python setup.py install`, the second command will redo all of the work completed in the first stage, so I removed the build phase from this package.